providers/ethereum-provider/src/EthereumProvider.ts

```diff
@@ -320,6 +324,60 @@ export class EthereumProvider implements IEthereumProvider {
     }
   }
 
+  public async authenticate(
+    params: AuthenticateParams,
+  ): Promise<AuthTypes.AuthenticateResponseResult | undefined> {
+    if (!this.signer.client) {
+      throw new Error("Provider not initialized. Call init() first");
+    }
+
+    this.loadConnectOpts({
+      chains: params?.chains,
+    });
+
+    try {
+      const result = await new Promise<AuthTypes.AuthenticateResponseResult>(
+        async (resolve, reject) => {
+          if (this.rpc.showQrModal) {
+            this.modal?.subscribeModal((state: { open: boolean }) => {
+              // the modal was closed so reject the promise
+              if (!state.open && !this.signer.session) {
+                this.signer.abortPairingAttempt();
+                reject(new Error("Connection request reset. Please try again."));
+              }
+            });
+          }
+          await this.signer
+            .authenticate({
+              ...params,
+              chains: this.rpc.chains,
+            })
+            .then((result: AuthTypes.AuthenticateResponseResult) => {
+              resolve(result);
+            })
+            .catch((error: Error) => {
+              reject(new Error(error.message));
+            });
+        },
+      );
+      if (!result) return;
```

Comment:
Should we add a log here if `result` is `undefined` to catch potential edge cases ?

Reply:
actually the condition was redundant as if result is not available, the promise will throw anyway